lib/validator.js

```diff
@@ -3,19 +3,28 @@
 const { Buffer } = require('buffer');
 const s = require('./signature');
 
+/**
+ *
+ * @param buffer {Buffer}
+ * @param fns {Array<function>}
+ * @returns {boolean|Error}
+ */
 const oneOf = (buffer, ...fns) => {
```

Comment:
ini sepertinya lebih simpel dengan foreach sebelumnya ya mas, 
kalau dengan for loop perlu memperkenalkan variabel baru, dalam hal ini `i`.
Plus kenapa ketika di bagian `if()` langsung `return`, tujuannya biar tidak ada `stackframe` yang sia sia, karena sudah bisa langsung jump keluar di fungsi tersebut.

di code sebelumnya juga tidak perlu ada variabel baru `valid`, baru ngeh juga.

sehingga lebih bagus seperti ini untuk fungsi `oneOf`

```javascript
const oneOf = (buffer, ...fns) => {
  for (const fn of fns) {
    if (typeof fn !== 'function') {
      throw new Error('second and next parameter should be function');
    }

    if (fn(buffer)) {
      return true;
    }
  }

  return false;
};
```

Reply:
> ini sepertinya lebih simpel dengan foreach sebelumnya ya mas, kalau dengan for loop perlu memperkenalkan variabel baru, dalam hal ini `i`. Plus kenapa ketika di bagian `if()` langsung `return`, tujuannya biar tidak ada `stackframe` yang sia sia, karena sudah bisa langsung jump keluar di fungsi tersebut.
> 
> di code sebelumnya juga tidak perlu ada variabel baru `valid`, baru ngeh juga.
> 
> sehingga lebih bagus seperti ini untuk fungsi `oneOf`
> 
> ```js
> const oneOf = (buffer, ...fns) => {
>   for (const fn of fns) {
>     if (typeof fn !== 'function') {
>       throw new Error('second and next parameter should be function');
>     }
> 
>     if (fn(buffer)) {
>       return true;
>     }
>   }
> 
>   return false;
> };
> ```

iya mas, baru ngeh kalau sudah kena return saat loop, kirain di luar scope loop, hehe

Reply:
yoi mas hehe, bakal irit stackframe